packages/analytics/__tests__/Analytics.test.ts

```diff
@@ -126,6 +152,26 @@ describe('Analytics test', () => {
 			);
 			expect(record_spyon).toBeCalled();
 		});
+
+		/**
+		 * Custom providers are a publically available feature via the `AnalyticsProvider` interface. Exercise caution when
+		 * updating these to ensure backwards compatibility.
+		 */
+		test('record with custom provider', async () => {
+			const analytics = new Analytics();
+			const customProvider = new TestCustomProvider();
+			const customProviderRecordSpy = jest.spyOn(customProvider, 'record');
+
+			analytics.addPluggable(customProvider);
+
+			const recordResponse = await analytics.record(
+				{ name: 'testEvent' },
+				'CustomProvider'
+			);
+
+			expect(customProviderRecordSpy).toBeCalled();
+			expect(recordResponse).toBeTruthy();
```

Comment:
Nit: Avoid using `toBeTruthy` when testing for boolean response
```suggestion
			expect(recordResponse).toBe(true);
```